examples/cli_workflow/README.md

```diff
@@ -0,0 +1,288 @@
+```
+
+### data
+
+Configure the `DistributedAnnDataCollection`. Here we validate `obs` columns that are used by the transforms and the model (`total_mrna_umis`):
+
+```diff
+<       filenames: null
+<       shard_size: null
+<       max_cache_size: 1
+<       obs_columns_to_validate: null
+---
+>       filenames: https://storage.googleapis.com/dsp-cellarium-cas-public/test-data/test_{0..3}.h5ad
+>       shard_size: 100
+>       max_cache_size: 2
+>       obs_columns_to_validate:
+>         - total_mrna_umis
+```
+
+Configure the `DataLoader`. `batch_keys` have to include all input arguments to the transforms and the model. For example, `NormalizeTotal`'s arguments are `x_ng` and `total_mrna_umis_n`, `Log1p`'s argument is `x_ng`, and `OnePassMeanVarStd`'s arguments are `x_ng` and `var_names_g`:
+
+```diff
+<   batch_keys: null
+<   batch_size: 1
+<   num_workers: 0
+---
+>   batch_keys:
+>     x_ng:
+>       attr: X
+>       convert_fn: cellarium.ml.utilities.data.densify
+>     var_names_g:
+>       attr: var_names
+>     total_mrna_umis_n:
+>       attr: obs
+>       key: total_mrna_umis
+>   batch_size: 100
+>   num_workers: 2
+```
+
+## `IncrementalPCA`
+
+Generate a default config file:
+
+```
+cellarium-ml incremental_pca fit --print_config > ipca_train_config.yaml
+```
+
+Below we highlight the changes made to the default configuration file.
+
+### train
+
+Change the number of devices and set the path for logs and weights:
+
+```diff
+<   devices: auto
+<   default_root_dir: null
+---
+>   devices: 2
+>   default_root_dir: /tmp/test_examples/ipca
+```
+
+### model
+
+Add `NormalizeTotal` and `Log1p`, and `ZScore` transforms. Note, that `mean_g`, `std_g`, and `var_names_g` of `ZScore` transform are loaded from the `OnePassMeanVarStd` checkpoint:
+
+```diff
+<   transforms: null
+---
+>   transforms:
+>     - class_path: cellarium.ml.transforms.NormalizeTotal
+>       init_args:
+>         target_count: 10_000
+>     - cellarium.ml.transforms.Log1p
+>     - class_path: cellarium.ml.transforms.ZScore
+>       init_args:
+>         mean_g:
+>           !CheckpointLoader
+>           file_path: /tmp/test_examples/onepass/lightning_logs/version_0/checkpoints/epoch=0-step=2.ckpt
+>           attr: model.mean_g
+>           convert_fn: null
+>         std_g:
+>           !CheckpointLoader
+>           file_path: /tmp/test_examples/onepass/lightning_logs/version_0/checkpoints/epoch=0-step=2.ckpt
+>           attr: model.std_g
+>           convert_fn: null
+>         var_names_g:
+>           !CheckpointLoader
+>           file_path: /tmp/test_examples/onepass/lightning_logs/version_0/checkpoints/epoch=0-step=2.ckpt
+>           attr: model.var_names_g
+>           convert_fn: numpy.ndarray.tolist
+```
+
+Set the number of components and mean correction option to true for `IncrementalPCA`:
+
+```diff
+<       n_components: null
+<       perform_mean_correction: false
+---
+>       n_components: 50
+>       perform_mean_correction: true
+```
+
```

Comment:
Again, add a sticky note that since we have z-scored the data, mean correction is not strictly necessary but its presence may help mitigate roundoff errors. It is a good exercise for the user to ascertain that the learned mean by IPCA is close to 0/

Reply:
I set`perform_mean_correction: true` by mistake. It should be `false` in this case because of the z-score!